pkg/dwarf/godwarf/type.go

```diff
@@ -1013,6 +1013,9 @@ func readType(d *dwarf.Data, name string, r *dwarf.Reader, off dwarf.Offset, typ
 	if err != nil {
 		goto Error
 	}
+	if typ == nil {
+		goto Error
```

Comment:
You should set err to something `err = fmt.Errorf("unexpected tag %v while reading type at offset %#x", e.Tag, e.Offset)`.

Is it possible to see the executable file that causes this?

Reply:
Done. 
About the executable, I think the main reason is changing CG_CFLAGS , I tried to add "-g -O0" but it didn't solve the issue. I really don't know why it happens. Maybe some C/C++ global variable is there? 
Interestingly that local vars works fine. So should be something related to global vars.

Reply:
This patch fixed the issue. I think this check is good to be there.

Reply:
This happens because we can't read a type fully. It doesn't happen with local variables because they just happen to not use the type we can't read. This change does prevent the segfault but we're still not reading one type. Hence why I'd like to see the executable.